src/cljs/vanilla/update_layout.cljs

```diff
@@ -1,4 +1,17 @@
-(ns vanilla.update-layout)
+(ns vanilla.update-layout
+  (:require
+    [ajax.core :as ajax :refer [GET POST]]))
+
+
+(defn save-layout [layout]
+  ;(prn "saving layout: " (clojure.core/pr-str layout))
+
+  (POST "/save-layout"
+        {:format          (ajax/json-request-format {:keywords? true})
+         :response-format (ajax/json-response-format {:keywords? true})
+         :params          {:widgets (clojure.core/pr-str layout)}       ;convert the whole layout struct to a string to preserve values
+         :handler         #(prn "Layout SAVED!")
+         :on-error        #(prn "ERROR saving the layout " %)}))
```

Comment:
here we can dispatch a message to re-frame that will cause a notification, with a message that we provide, will spread for 3 seconds (?) or should we let the user interact? I'd hate to force a modal since there really isn't anything the user can do if a save fails anyway.